certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -377,15 +502,19 @@ def choose_redirect_vhost(self, target_name, port, create_if_no_match=False):
             when there is no match found. If we can't choose a default, raise a
             MisconfigurationError.
 
-        :returns: vhost associated with name
-        :rtype: :class:`~certbot_nginx.obj.VirtualHost`
+        :returns: vhosts associated with name
+        :rtype: list of :class:`~certbot_nginx.obj.VirtualHost`
 
         """
-        matches = self._get_redirect_ranked_matches(target_name, port)
-        vhost = self._select_best_name_match(matches)
-        if not vhost and create_if_no_match:
-            vhost = self._vhost_from_duplicated_default(target_name, port=port)
-        return vhost
+        if _wildcard_domain(target_name):
+            # Ask user which VHosts to enhance.
+            vhosts = self._choose_vhosts_wildcard(target_name, prefer_ssl=False)
```

Comment:
Two things:

1. This method ignores the port number given to this function. Are we OK with this?
2. With `prefer_ssl`, we'll still return SSL vhosts if no plaintext vhosts exist. Is this correct?

Reply:
1. Not ideal because they could theoretically pick a bad one. I'll fix it.
2. Yes -- the test is "is it listening on 80", not "is it not listening on 443 ssl". And it's really just a filter to cut down the number of vhosts the poor users have to sift through; too many is fine.

and, done.